src/controllers/admin/survey-administration-controller.ts

```diff
@@ -0,0 +1,161 @@
+import { ValidationError } from "yup"
+import { type Request, type Response } from "express"
+import * as SurveyAdministrationService from "../../services/survey-administration-service"
+import CustomError from "../../utils/custom-error"
+import { createSurveyAdministrationSchema } from "../../utils/validation/survey-administration-schema"
+
+/**
+ *List survey administrations
+ @param req.body.name - Filter by name 
+ @param req.body.status - Filter by status 
+ */
+export const index = async (req: Request, res: Response) => {
+  try {
+    const { name, status } = req.query
+    const survey = await SurveyAdministrationService.getAllByFilters(
+      name as string,
+      status as string
+    )
+    res.json(survey)
+  } catch (error) {
+    res.status(500).json({ message: "Something went wrong" })
+  }
+}
+
+/**
+ * Get a specific survey administration by ID.
+ * @param req.params.id - The unique ID of the survey administration
+ */
+export const show = async (req: Request, res: Response) => {
+  try {
+    const { id } = req.params
+    const survey = await SurveyAdministrationService.getById(parseInt(id))
+    res.json(survey)
+  } catch (error) {
+    res.status(500).json({ message: "Something went wrong" })
+  }
+}
+
+/**
+ * Store a new survey administration
+ * @param req.body.name - Name
+ * @param req.body.survey_start_date - Survey start date
+ * @param req.body.survey_end_date - Survey end date
+ * @param req.body.survey_template_id - Survey template id
+ * @param req.body.remarks - Survey remarks
+ * @param req.body.email_subject - Email subject
+ * @param req.body.email_content - Email content
+ * @param req.body.status - Status
+ */
+export const store = async (req: Request, res: Response) => {
+  try {
+    const {
+      name,
+      survey_start_date,
+      survey_end_date,
+      survey_template_id,
+      remarks,
+      email_subject,
+      email_content,
+    } = req.body
+    await createSurveyAdministrationSchema.validate({
+      name,
+      survey_start_date,
+      survey_end_date,
+      survey_template_id: parseInt(survey_template_id),
+      remarks,
+      email_subject,
+      email_content,
+    })
+
+    const newSurvey = await SurveyAdministrationService.create({
+      name: name as string,
+      survey_start_date: new Date(survey_start_date),
+      survey_end_date: new Date(survey_end_date),
+      survey_template_id: parseInt(survey_template_id),
+      remarks,
+      email_subject,
+      email_content,
+    })
```

Comment:
Kindly add status: SurveyAdministrationStatus.Draft here to auto set the status to Draft